docs/coding-convention.md

```diff
@@ -0,0 +1,55 @@
+
+# Coding Convention
+
+## C codes (.c sources)
+
+All C codes of NNTrainer are required to use K&R with 2-space indenting. Just follow what's already there and you'll be fine. We only require .c files to be indented, headers may be indented differently for better readability. Please use spaces for indenting, not tabs, even in header files.
+
+When you push your commits, apply `indent.sh` to .c files before you submit a commit with style change.
+If there is a change due to code style issues, make two separate commits: (Please do not include other codes' style change in the same commit)
+- commit with style change only
+- commit with your code change only (i.e., contents only).
+
+```
+$ ./tools/development/indent.sh <file-name>
+```
+
+## C headers (.h)
+
+You may indent differently from what indent.sh does. You may also break the 80-column rule with header files.
+
+Except the two, you are required to follow the general coding styles mandated by gst-indent
+
+## C++ files (.cpp)
+
+Use .h for headers and .cpp for source.
+You have to use clang-format with the given [.clang-format](../.clang-format) file
+
+
+## Other files
+
+- [Java] TBD
+- [Python] TBD
+- [Bash] TBD
+
+
+# File Locations
+
+## Directory structure of nnstreamer.git
+
+- **api**: API definitions and implementations
+    - **capi**: C-APIs (Tizen and others)
+- **debian**: Debian/Ubuntu packaging files
+- **docs**: Documentations
+- **nntrainer**: All core NNTrainer codes are located here
+- **nnstreamer**: NNStreamer sub-filter codes for NNTrainer
+- **jni**: Android/Java build scripts.
+- **Applications**: Examples for NNtrainer
+- **packaging**: Tizen RPM build scripts. OpenSUSE/Redhat Linux may reuse this.
+- **test**: Unit test cases. We have GTEST test cases. There are subdirectories, which are groups of unit test cases.
+- **tools**: Various developmental tools and scripts of NNTrainer.
+
+## Related git repositories
+
+- [NNStreamer](https://github.com/nnstreamer/nnstreamer)
```

Comment:
also add https://github.com/nnstreamer/api
this will soon become the API repo.

Reply:
I think later.